draft-ietf-lake-edhoc.md

```diff
@@ -650,59 +654,52 @@ where salt = 0x (zero-length byte string).
 
 The pseudo-random key PRK_3e2m is used to produce a MAC in message_2 and to encrypt message_3. PRK_3e2m is derived as follows:
 
-If the Responder authenticates with a static Diffie-Hellman key, then PRK_3e2m = Extract( PRK_2e, G_RX ), where G_RX is the ECDH shared secret calculated from G_R and X, or G_X and R (the Responder's private authentication key, see {{auth-keys}}), else PRK_3e2m = PRK_2e.
-
-### PRK_3m
+If the Responder authenticates with a static Diffie-Hellman key, then PRK_3e2m = Extract( SALT_3e2m, G_RX ), where
 
-The pseudo-random key PRK_3m is used to produce a MAC in message_3. PRK_3m is derived as follows:
+* SALT_3e2m is derived from PRK_2e, see {{expand}}, and
+* G_RX is the ECDH shared secret calculated from G_R and X, or G_X and R (the Responder's private authentication key, see {{auth-keys}}),
 
-If the Initiator authenticates with a static Diffie-Hellman key, then PRK_3m = Extract( PRK_3e2m, G_IY ), where G_IY is the ECDH shared secret calculated from G_I and Y, or G_Y and I (the Initiator's private authentication key, see {{auth-keys}}), else PRK_3m = PRK_3e2m.
+else PRK_3e2m = PRK_2e.
 
-### PRK_out
+### PRK_4e3m
 
-The pseudo-random key PRK_out is used to derive application specific data and keying material to encrypt message_4. PRK_out is derived as follows:
+The pseudo-random key PRK_4e3m is used to produce a MAC in message_3 and to encrypt message_4. PRK_4e3m is derived as follows:
 
-PRK_out = Extract(PRK_3m, TH_4)
+If the Initiator authenticates with a static Diffie-Hellman key, then PRK_4e3m = Extract( SALT_4e3m, G_IY ), where
 
-The transcript hash TH_4 is a CBOR encoded bstr and the input to the hash function is a CBOR Sequence.
+* SALT_4e3m is derived from PRK_3e2m, see {{expand}}, and
+* G_IY is the ECDH shared secret calculated from G_I and Y, or G_Y and I (the Initiator's private authentication key, see {{auth-keys}}),
 
-~~~~~~~~~~~
-   TH_4 = H( TH_3, PLAINTEXT_3 )
-~~~~~~~~~~~
+else PRK_4e3m = PRK_3e2m.
 
-where H() is the EDHOC hash algorithm in the selected cipher suite.
 
 ## Expand {#expand}
 
-The keys, IVs and MACs used in EDHOC are derived from the PRKs using Expand, and instantiated with the EDHOC AEAD algorithm in the selected cipher suite.
+The keys, IVs, salts, and MACs used in EDHOC are derived from the PRKs using Expand.
 
 ~~~~~~~~~~~~~~~~~~~~~~~
-   OKM = EDHOC-KDF( PRK, transcript_hash, label, context, length )
+   OKM = EDHOC-KDF( PRK, label, context, length )
        = Expand( PRK, info, length )
 ~~~~~~~~~~~~~~~~~~~~~~~
 
```

Comment:
OKM is now undefined. The text "output keying material (OKM)" was deleted in an earlier paragraph. Maybe OKM can be completely removed? But if is is used it has to be defined.

Reply:
OKM was defined in an earlier section, now also mentioned here.